File: ABC/79/b.c
/* TLE ...... */

#include <stdio.h>
#include <string.h>

typedef long long int lli;

lli lucas_number(lli N);

int main(void) {
  // input
  lli N; scanf("%lld", &N);

  // output
  printf("%lld\n", lucas_number(N));
  return 0;
}


lli lucas_number(lli N) {
  if (N == 0) {
    return 2;
  }
  else if (N == 1) {
    return 1;
  }
  else if (N >= 2) {
    return lucas_number(N-1) + lucas_number(N-2);
  }
}

File: ABC/79/b.py
### TLE ......

def lucas_number(N: int) -> int:
  if N == 0:
    return 2
  elif N == 1:
    return 1
  elif N >= 2:
    return lucas_number(N-1) + lucas_number(N-2)


def main():
  # input
  N = int(input())

  # output
  print(lucas_number(N))

if __name__ == "__main__":
    main()
